src/TSMapEditor/StreamHelpers.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.IO;
+using System.Text;
+
+namespace TSMapEditor
+{
+    public class StreamHelperReadException : Exception
+    {
+        public StreamHelperReadException(string message) : base(message)
+        {
+        }
+    }    
+
+    public static class StreamHelpers
+    {
+        public static int ReadInt(Stream stream)
+        {
+            byte[] buffer = new byte[8];
+
+            if (stream.Read(buffer, 0, 4) != 4)
+                throw new StreamHelperReadException("Failed to read integer from stream: end of stream");
+
+            return BitConverter.ToInt32(buffer, 0);
```

Comment:
`new` allocates heap on the stack, which is expensive. Avoid it if possible by using `Span`s and `stackalloc` instead.

```suggestion
            Span<byte> buffer = stackalloc byte[4];

            if (stream.Read(buffer) != sizeof(int))
                throw new StreamHelperReadException("Failed to read integer from stream: end of stream");

            return BitConverter.ToInt32(buffer);
```

Adjust the other functions accordingly.